### ISSUE TITLE
Don't show (Edit Visually) button for users without edit permission, even if a user knows how to

### ISSUE THREAD

**Issue by hazho** [hazho/Wagtail-GrapesJS#4]
cases:

1. visitors site view:
   while the user is authenticated and only if the user has the edit permission over the page object visited, the (Edit Visually) should be listed in UserBar menu, but not for any other users.
2.  Page Editor Site view(in admin site):
   users with edit permission can see and perform editing the page vissually, through the action buttons (a custom action button needed)

**Comment by hazho** [comment]
while we are registering a userBar menu item for the GrapesJS as (Edit Visually), we can specify this button to be rendered for whom, as you mentioned, we can look at the built-in (edit this page) userBar menu item and implement this button as so, except the actions to be handled for loading GrapesJS resources (JS/CSS) or just simply initiating it (means the lib js and css will always be loaded forthe editor users at visitor's site view)

**Comment by itzomen** [comment]
> while we are registering a userBar menu item for the GrapesJS as (Edit Visually), we can specify this button to be rendered for whom, as you mentioned, we can look at the built-in (edit this page) userBar menu item and implement this button as so, except the actions to be handled for loading GrapesJS resources (JS/CSS) or just simply initiating it (means the lib js and css will always be loaded forthe editor users at visitor's site view)

Yeah, we can have it loaded by default for editors, for now then later on optimize it if it has major performance issues
